### PR TITLE
Moving createEncoder to ValidationTest

### PR DESCRIPTION
**Pull request by egalli**
Follow-up on https://github.com/gpuweb/cts/pull/286#discussion_r487357285

@kainino0x is this what you were thinking?

### CODE REVIEW

File: src/webgpu/api/validation/validation_test.ts [hunk 1]
```diff
@@ -148,6 +156,77 @@ export class ValidationTest extends GPUTest {
...
+  createEncoder(encoderType: 'render pass'): GPURenderPassEncoder & FinishableEncoder;
+  createEncoder(encoderType: 'compute pass'): GPUComputePassEncoder & FinishableEncoder;
+  createEncoder(encoderType: 'render bundle'): GPURenderBundleEncoder & FinishableEncoder;
+  createEncoder(
+    encoderType: 'render pass' | 'render bundle'
+  ): (GPURenderPassEncoder | GPURenderBundleEncoder) & FinishableEncoder;
+  createEncoder(
+    encoderType: 'compute pass' | 'render pass' | 'render bundle'
+  ): GPUProgrammablePassEncoder & FinishableEncoder;
+  createEncoder(encoderType: EncoderType): Encoder & FinishableEncoder;
```
Comment (kainino0x) [inline src/webgpu/api/validation/validation_test.ts:169]: Nice type preservation with these overloads!

**Review by kainino0x — APPROVED** [review]
Yup, this is exactly the abstraction I had in mind!

The only difference with what I had in mind was I was thinking of returning something like `class CommandBufferMaker<E extends Encoder> { readonly encoder: E; finish(): GPUCommandBuffer; }`. It would be a little more verbose (`maker.encoder.pushDebugGroup()`) but maybe a little clearer especially to contributors who are less familiar with TypeScript. Either is okay with me though; WDYT?

**Comment by egalli** [comment]
Sounds good to me. I was never a huge fan of modifying native types/casting through unknown. Regarding the extra verbosity, you can always destructure the type. 😄

**Review by kainino0x — APPROVED** [review]
> Regarding the extra verbosity, you can always destructure the type. 😄

Very true, good call!

**Pull request merged**